solr/prometheus-exporter/src/java/org/apache/solr/prometheus/scraper/SolrScraper.java

```diff
@@ -130,29 +134,35 @@ protected MetricSamples request(SolrClient client, MetricsQuery query) throws IO
       zkHostLabelValue = ((CloudSolrClient) client).getClusterStateProvider().getQuorumHosts();
     }
 
-    QueryRequest queryRequest = new QueryRequest(query.getParameters());
-    queryRequest.setPath(query.getPath());
-
-    NamedList<Object> queryResponse;
+    SolrRequestType requestType = SolrRequestType.QUERY;
```

Comment:
I think these ~9 lines would read clearer as an if-else, with initializing a request two different ways.  Rather than declaring a bunch of vars that are components to a request.

Reply:
Done

Reply:
Done